oaknational/Oak-Web-Application
# LESQ-1044, LESQ-1047, LESQ-1049: Unit Listing page filters

## Description

Music year: {owa_music_year}

- Adds Subject Category Filters 
- Adds Year Group Filters
- Amends Unit Listing titles 
- Updates filterUnits function to add years and category options
- Adds 'No results' text when no results are returned

## Issue(s)

## Acceptance Criteria:

LESQ-1044 - Subject Category Filters
- [x]  KS2 english and sciences are where they are currently used
- [x]  KS4 combined science should have all subject categories, KS4 physics/bio/chem shouldn’t display subject category
- [x]  Displayed above threads in the side bar
- [x]  Radio buttons
- [x]  New title “Filters” with the sub-categories “Categories” and “Threads” (same structure as what we have in search)
- [x]  Filters the list of displayed units to whichever category was selected
- [x]  The numbering on the new list goes from 1 onwards (its a new order, don’t use the original ordering of the units)
- [x]  When there no results, display “No results. Please try removing some filters.”
- [x]  Browse refined event should fire when a subject category is selected
- [ ]  Add subject category icons to radio buttons - Not utilised radio buttons for this 
- [x]  Query param for when subject category is selected (?category=)
- [x]  Displayed above threads in the mobile drop down
- [x]  Selecting a subject category means no legacy content will be displayed

LESQ-1049 - Year group filters
- [x]  Year group filter added above subject categories
- [x]  Radio buttons
- [x]  Applies to both new and legacy content
- [x]  Add to mobile dropdown
- [x]  Query param ?year=

LESQ-1047 - Unit listing titles
- [x]  When a subject category is selected, then the title at the top of the unit list changes from `[Subject] units` to `[Subject category] units`
- [x]  When all is selected (by default), display `[Subject] units`
- [x]  changes to the component:
- [x]  using heading-5 instead of heading-4
- [x]  update colour to use design token: text-primary

Fixes #LESQ-1044, LESQ-1047, LESQ-1049

## How to test

1. Go to https://deploy-preview-2768--oak-web-application.netlify.thenational.academy /teachers/programmes/combined-science-secondary-ks4-foundation-edexcel/units or https://deploy-preview-2768--oak-web-application.netlify.thenational.academy/teachers/programmes/english-primary-ks2/units
2. Click on unit filters, year or subject category
3. You should see URL changes, units are filtered based on chosen category, filter is highlighted

#### N.B. Currently threads is being refactored, so the page refreshes when clicking on a thread, this will be fixed with this PR https://github.com/oaknational/Oak-Web-Application/pull/2784

## Screenshots


How it should now look:

KS4 - Combined Science
![Screenshot 2024-09-16 at 14 26 48](https://github.com/user-attachments/assets/4ff506b3-6d90-4a1b-9c30-548b465d6157)


https://github.com/user-attachments/assets/79a91845-9edd-4e8e-a6db-17358e8bde5f



KS2 - English

![Screenshot 2024-09-16 at 14 27 18](https://github.com/user-attachments/assets/b17ebd73-729d-49ee-beb9-9ba1eb103de4)


https://github.com/user-attachments/assets/91aadf6a-adfc-4252-b639-2a526937ee56



## Checklist
